### PR TITLE
add port accessors and modifiers for TCP/UDP

### PR DESCRIPTION
**Pull request by spurserh**

### CODE REVIEW

File: waterutil/tun_ipv4.go [hunk 1]
```diff
@@ -28,11 +28,42 @@ func IPv4Source(packet []byte) net.IP {
 	return net.IPv4(packet[12], packet[13], packet[14], packet[15])
 }
 
+func SetIPv4Source(packet []byte, source net.IP) {
+	copy(packet[12:16], source)
```
Comment (songgao) [inline waterutil/tun_ipv4.go]: (same for `SetIPv4Destination`,) Maybe use [`net.IP.To4()`](https://golang.org/pkg/net/#IP.To4) to check if `source` is indeed an IPv4 address? Or perhaps at least replace `source` with `source.To4()` so it results in a runtime error when non-IPv4 address is used?

**Comment by songgao** [comment]
Thanks @spurserh !

I'm also working on adding IP support to [a separate repo](https://github.com/songgao/packets) just for parsing and constructing packets, which I'm thinking about replacing `waterutil` with. Once it's done, would you be interested in reviewing it and perhaps adding some transport layer support?

**Comment by spurserh** [comment]
Hey @songgao : Thanks for your suggestion. 

I am relatively new to golang, but sure, I am willing to help with /packets. Let me know!

**Comment by songgao** [comment]
Thanks @spurserh !

**Pull request merged**